### PR TITLE
Added comments in server.py

### PR DESCRIPTION
**Pull request by Priprocks**
**Status (please check what you already did)**:
- [ ] made PR ready for code review
- [ ] added some tests for the functionality
- [x] updated the documentation
- [ ] updated the changelog

### CODE REVIEW

File: rasa_core/server.py [hunk 1]
```diff
@@ -146,25 +146,26 @@ def _create_agent(
...
-               interpreter=None,
-               loglevel="INFO",
-               logfile="rasa_core.log",
-               cors_origins=None,
-               action_factory=None,
-               auth_token=None,
-               tracker_store=None
+def create_app(model_directory,           # type: Text
+               interpreter=None,          # type: Optional
+                                        # [Union(text,NL interpreter)]
```
Comment (tmbo) [inline rasa_core/server.py]: this isn't a proper type hint, should be `Union[Text, NLI, None]` with a

```
if typing.TYPE_CHECKING:
    from rasa_core.interpreter import NaturalLanguageInterpreter as NLI
```
at the top (after the imports).

additionally you need to add 
```
from typing import Union
import typing
```
to the imports

Comment (tmbo) [reply rasa_core/server.py]: please also don't align the type hints but rather put them two spaces after the argument.

File: rasa_core/server.py [hunk 2]
```diff
@@ -146,25 +146,26 @@ def _create_agent(
...
     app = Flask(__name__)
     CORS(app, resources={r"/*": {"origins": "*"}})
-
+    # Setting up logfile
     utils.configure_file_logging(loglevel, logfile)
 
     if not cors_origins:
         cors_origins = []
-
+    type(cors_origins)
```
Comment (tmbo) [inline rasa_core/server.py]: I guess this is not needed

Comment (Priprocks) [reply rasa_core/server.py]: Okay sure! I'm making the necessary changes as you said

**Comment by Priprocks** [comment]
Did the changes as you asked. Thank you for helping me out! :)

File: rasa_core/server.py [hunk 3]
```diff
@@ -146,25 +152,24 @@ def _create_agent(
...
-def create_app(model_directory,
-               interpreter=None,
-               loglevel="INFO",
-               logfile="rasa_core.log",
-               cors_origins=None,
-               action_factory=None,
-               auth_token=None,
-               tracker_store=None
+def create_app(model_directory,  # type: Text
+               interpreter=None,  # type: Optional[Union(text,NL interpreter)]
```
Comment (tmbo) [inline rasa_core/server.py]: should be `# type: Union[Text, NLI, None]`

**Comment by tmbo** [comment]
Thanks :rocket:

**Pull request merged**